QModManager/Patching/ManifestValidator.cs

```diff
@@ -145,15 +145,38 @@ public void CheckRequiredMods(QMod mod)
             }
 
             mod.RequiredMods = requiredMods.Values;
-            if (Logger.DebugLogsEnabled && requiredMods.Count > 0)
+            if (Logger.DebugLogsEnabled)
             {
-                string msg = $"{mod.Id} has required mods: ";
-                foreach (var required in requiredMods.Values)
+                if (requiredMods.Count > 0)
                 {
-                    msg += $"{required.Id} ";
+                    string msg = $"{mod.Id} has required mods: ";
+                    foreach (var required in requiredMods.Values)
+                    {
+                        msg += $"{required.Id} ";
+                    }
+
+                    Logger.Debug(msg);
                 }
+                if (mod.LoadBeforePreferences.Count > 0)
+                {
+                    string msg = $"{mod.Id} should load before: ";
+                    foreach (var id in mod.LoadBeforePreferences)
+                    {
+                        msg += $"{id} ";
+                    }
+
+                    Logger.Debug(msg);
+                }
+                if (mod.LoadAfterPreferences.Count > 0)
+                {
+                    string msg = $"{mod.Id} should load after: ";
+                    foreach (var id in mod.LoadAfterPreferences)
+                    {
+                        msg += $"{id} ";
+                    }
```

Comment:
Consider a small refactor of this repeated code.

```CS
// could be private to the scope of the if-block
string GetModList(string msg, IEnumerable<string> list)
{
    foreach (var id in list)
        msg += $"{id} ";

    return msg;
}

if (requiredMods.Count > 0)
{
    string msg = $"{mod.Id} should load before: ";
    Logger.Debug(GetModList(msg, requiredMods.Values);
}
```

Reply:
Doesn't have to be that exact code.  
Here's an alternative
```CS
string GetModList(IEnumerable<string> list)
{
    string msg = string.Empty;
    foreach (var id in list)
        msg += $"{id} ";

    return msg;
}

if (requiredMods.Count > 0)
{
    Logger.Debug(($"{mod.Id} should load before: " + GetModList(requiredMods.Values);
}
```

Reply:
4518363

Reply:
Assuming this change works and the `<cast>` returns the Mod ID like we want, then this is fine.  
Alternatively, you can just revert the `RequiredMods` list back to its own method or just undo this refactor as I forgot that not all 3 collections were just string IDs.